### ISSUE TITLE
iOS - Logic - Delete Timed Activity

### ISSUE THREAD

**Issue by ceceliacreates** [WomenWhoCode/WWCodeMobile#80]
Add logic to app to delete timed activities.

- When tapping the "ok" option on the delete confirmation popup
- Existing entry should be deleted

<img width="865" alt="Screenshot 2024-01-24 at 6 13 22 PM" src="https://github.com/WomenWhoCode/WWCodeMobile/assets/40367173/8365d594-7a07-4d9c-83ac-df9b33fe1d0d">

**Comment by pelumy** [comment]
@clc80 and @devanshimodha, I would like to work on this. Doesn't the create task logic block this issue?
Also, do we want to use a data persistence like coreData?

**Comment by devanshimodha** [comment]
@pelumy we decided to use SwiftData but temporarily it is okay implement in-memory dictionaries / arrays as well

**Comment by ceceliacreates** [comment]
Hi @pelumy is there any update on this issue?

**Comment by pelumy** [comment]
Hi @clc80, I was waiting for the Add/Edit Logic to be completed before starting this.
Reason being that, the task needs to be passed from the ListScreen to the DetailScreen. Especially in the case of Editing, I don't want a situation where we both try to pass the task to the Details Screen and the other person has to remove their own implementation.

What do you think about this? Should I still work on this regardless?

**Comment by clc80** [comment]
> Hi @clc80, I was waiting for the Add/Edit Logic to be completed before starting this. Reason being that, the task needs to be passed from the ListScreen to the DetailScreen. Especially in the case of Editing, I don't want a situation where we both try to pass the task to the Details Screen and the other person has to remove their own implementation.
> 
> What do you think about this? Should I still work on this regardless?

@pelumy we can go about this in two ways. We can merge this PR as is and then open up another issue to pass the info. Or we can continue to wait. 

The only issue with waiting for the Add/Edit logic is that we don't know when it will be incorporated into the project, so you may be waiting for a while.

**Comment by pelumy** [comment]
@clc80, thank you for the feedback. I will create a new PR for this issue and work on passing the task to the details list next. Thanks for your patience.

**Comment by clc80** [comment]
@pelumy Let us know here or on slack if you have anymore questions.
